modules/giapi/src/main/scala/giapi/client/GiapiStatusDb.scala

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package giapi.client
+
+import cats.Applicative
+import cats.implicits._
+import cats.effect.Sync
+import cats.effect.Resource
+import cats.effect.Effect
+import cats.effect.ConcurrentEffect
+import cats.effect.implicits._
+import edu.gemini.aspen.giapi.status.StatusHandler
+import edu.gemini.aspen.giapi.status.StatusItem
+import edu.gemini.aspen.giapi.statusservice.StatusHandlerAggregate
+import edu.gemini.jms.activemq.provider.ActiveMQJmsProvider
+import fs2.concurrent.Queue
+import fs2.Stream
+
+/////////////////////////////////////////////////////////////////
+// Links status streaming with the giapi db
+/////////////////////////////////////////////////////////////////
+trait GiapiStatusDb[F[_]] {
+  def value(i: String): F[Option[StatusValue]]
+  def close: F[Unit]
+}
+
+object GiapiStatusDb {
+  private def dbUpdate[F[_]: Applicative, A](db:   GiapiDb[F],
+                                             name: String,
+                                             a:    A): F[Unit] =
+    a match {
+      case a: Int =>
+        db.update[Int](name, a)
```

Comment:
You probably don't need explicit type arguments here. Try `db.update(name, a)` … kind of weird because it looks like duplicated code but it's not.

Reply:
So, because `A` is unconstrained and doesn't appear in the return type and we're typecasing in the `match` I think it would express the meaning a little more clearly if we get rid of `A` and write `a: Any` in the signature.

Reply:
won't WartRemover balk at `Any`?

Reply:
Mmm possibly. But we're using `Any` in the `match` so I think we should be open about it.